### PR TITLE
Add scripts to create platform identities for dev

### PR DESCRIPTION
**Pull request by niontive**
### Which issue this PR addresses:

Fixes https://issues.redhat.com/browse/ARO-8997

### What this PR does / why we need it:

Platform identities will be used for MIWI local dev cluster.

### Test plan for issue:

I ran the scripts

### Is there any documentation that needs to be updated for this PR?

Yes - see PR

### CODE REVIEW

**Review by kimorris27 — DISMISSED** [review]
E2E failed, but it isn't relevant to this PR.

**Comment by kimorris27** [comment]
I wrote a ticket for the E2E flake that occurred: https://issues.redhat.com/browse/ARO-9445

**Review by bitoku — COMMENTED** [review]
It lacks error handling where it calls az commands.
I think `set -e` is enough.

File: hack/devtools/local_dev_env.sh [hunk 1]
```diff
@@ -7,6 +7,71 @@
 # The steps here are the ones defined in docs/deploy-development-rp.md
 # We recommend to use this script after you understand the steps of the process, not before.
 
+PLATFORM_WORKLOAD_IDENTITY_ROLE_SETS='[
+    {
+        "openShiftVersion": "4.14",
+        "platformWorkloadIdentityRoles": [
```
[the remark this answers is not in the document]
Comment (niontive) [reply hack/devtools/local_dev_env.sh:14]: I added an export for the role sets variable to `env`

`openShiftVersion` is used in this command: https://github.com/Azure/ARO-RP/blob/master/cmd/aro/update_role_sets.go, so we'll use it during testing.

**Review by kimorris27 — APPROVED** [review]

File: hack/devtools/local_dev_env.sh [hunk 2]
```diff
@@ -1,4 +1,5 @@
 #!/bin/bash
+set -e
```
Comment (stevekuznetsov) [inline hack/devtools/local_dev_env.sh:2]: Unless we have a specific reason not to, best to have the full preamble:

```sh
set -o errexit
set -o nounset
set -o pipefail
```

File: hack/devtools/msi.sh [hunk 3]
```diff
@@ -1,7 +1,17 @@
 #!/bin/bash
 
-# This script creates a mock MSI object to use for local development
-# We use a service principal and certificate as the mock object
+# This script creates a mock cluster MSI object and platform identities to use for local development
+# We use a service principal and certificate as the mock cluster MSI
+
+if [[ -z "$AZURE_SUBSCRIPTION_ID" ]]; then
```
Comment (stevekuznetsov) [inline hack/devtools/msi.sh:6]: Couple small nits that are likely best solved by turning on `shellcheck` - this file also is missing the error preamble, and it's best to wrap variable names in braces and handle the missing case, e.g. `"${AZURE_SUBSCRIPTION_ID:-}"`

Comment (niontive) [reply hack/devtools/msi.sh]: Thanks for pointing me to shellcheck - I turned it on in my IDE!

I found some other linting errors through shell check - pushed changes for those as well.

File: hack/devtools/local_dev_env.sh [hunk 4]
```diff
@@ -7,6 +10,71 @@
...
+                "roleDefinitionName": "Azure RedHat OpenShift Storage Operator Role",
+                "roleDefinitionId": "/providers/Microsoft.Authorization/roleDefinitions/5b7237c5-45e1-49d6-bc18-a1f62f400748",
+                "serviceAccounts": [
+                    "openshift-cluster-csi-drivers:azure-disk-csi-driver-operator",
+                    "openshift-cluster-csi-drivers:azure-disk-csi-driver-controller-sa"
+                ]
+            },
+            {
+                "operatorName": "NetworkOperator",
+                "roleDefinitionName": "Azure RedHat OpenShift Network Operator Role",
```
Comment (cadenmarchese) [inline hack/devtools/local_dev_env.sh:46]: Nit - we made some corrections to the `roleDefinitionName` names in RP-Config, so we should make them here as well: https://msazure.visualstudio.com/AzureRedHatOpenShift/_git/RP-Config/pullrequest/10029612

File: hack/devtools/local_dev_env.sh [hunk 5]
```diff
@@ -55,21 +123,107 @@ get_mock_msi_tenantID() {
...
+    az role assignment create --assignee-object-id "${objectId}" --assignee-principal-type "ServicePrincipal" --role "${roleId}"  --scope "${scope}" --output json
+    echo ""
+}
+
+create_platform_identity_and_assign_role() {
+    local operatorName="${1}"
+    local roleDefinitionId="${2}"
+    local identityName="aro-${operatorName}"
+    
+    identity=$(az identity show --name "${identityName}" --resource-group "${RESOURCEGROUP}" --subscription "${AZURE_SUBSCRIPTION_ID}" --output json 2>/dev/null)
```
Comment (kimorris27) [inline hack/devtools/local_dev_env.sh:170]: It seems that the error handling stuff you added in is going to require some changes throughout the script. I tried running the latest version and it exits after this line with a `ResourceNotFound` error.

**Review by kimorris27 — CHANGES_REQUESTED** [review]

File: hack/devtools/local_dev_env.sh (hunk 5 above)
Comment (niontive) [reply hack/devtools/local_dev_env.sh:170]: Thanks! I'll run this again via `local_dev_env.sh` and `msi.sh` to verify.

**Review by kimorris27 — APPROVED** [review]

File: hack/devtools/msi.sh [hunk 6]
```diff
@@ -13,11 +26,15 @@ if [[ -z "$sp" ]]; then
...
+cluster_msi_role_assignment "${mockClientID}"
+
 # Print the extracted values
-echo "Client ID: $clientID"
-echo "Tenant ID: $tenantID"
-echo "Base64 Encoded Certificate: $base64EncodedCert"
\ No newline at end of file
+echo "Cluster MSI Client ID: $mockClientID"
+echo "Cluster MSI Tenant ID: $mockTenantID"
+echo "Cluster MSI Base64 Encoded Certificate: $base64EncodedCert"
```
Comment (SudoBrendan) [inline hack/devtools/msi.sh:39]: nit: This isn't a change you introduced, but if we're here anyways... If I'm reading all this correctly - this is a private key? It's best to not print secrets to STDOUT even if it's obfuscated by base64 (encoding != encrypting).

**Review by SudoBrendan — APPROVED** [review]
Ultimately OK to merge, although I'd prefer if we didn't print secrets to terminal, ever.

**Review by rhamitarora — APPROVED** [review]

**Review by cadenmarchese — APPROVED** [review]

**Pull request merged**